eslint.config.mjs

```diff
@@ -0,0 +1,80 @@
+import path from "node:path"
+import tseslint from "typescript-eslint"
+import url from "node:url"
+import unusedImports from "eslint-plugin-unused-imports"
+
+const __dirname = path.dirname(url.fileURLToPath(import.meta.url))
+
+export default tseslint.config(
+    // register plugins
+    {
+        plugins: {
+            ["@typescript-eslint"]: tseslint.plugin,
+            ["@unused-imports"]: unusedImports,
+        },
+    },
+
+    // add files and folders to be ignored
+    {
+        ignores: [
+            "**/*.js",
+            "**/*.spec.ts",
+            "eslint.config.mjs",
+            "vitest.config.ts",
+            ".github/*",
+            ".yarn/*",
+            ".testing/*",
+            "dist/*",
+            "docs/*",
+            "scripts/*",
+        ],
+    },
+
+    tseslint.configs.all,
+
+    {
+        languageOptions: {
+            parserOptions: {
+                projectService: true,
+                tsconfigRootDir: __dirname,
+            },
+        },
+
+        rules: {
+            // override stylisticTypeChecked
+            "@typescript-eslint/no-empty-function": ["error", {allow: ["arrowFunctions"]}],
+            "@typescript-eslint/no-inferrable-types": "off",
+            "@typescript-eslint/typedef": [
+                "error",
+                {parameter: true, memberVariableDeclaration: true},
+            ],
+            "@typescript-eslint/consistent-generic-constructors": ["error", "type-annotation"],
+            "@typescript-eslint/prefer-optional-chain": "off",
+
+            // override strictTypeChecked
+            "@typescript-eslint/no-unused-vars": "off",
+            "@typescript-eslint/no-extraneous-class": "off",
+
+            "@typescript-eslint/prefer-readonly": "error",
+            "@typescript-eslint/switch-exhaustiveness-check": "error",
+
+            "@unused-imports/no-unused-imports": "error",
+            "no-duplicate-imports": "error",
+
+            "@typescript-eslint/no-magic-numbers": "off",
+            "@typescript-eslint/no-unsafe-type-assertion": "off",
+            "@typescript-eslint/prefer-readonly-parameter-types": "off",
+            "@typescript-eslint/member-ordering": "off",
+            "@typescript-eslint/method-signature-style": "off",
+            "@typescript-eslint/prefer-destructuring": "off",
+            "@typescript-eslint/strict-boolean-expressions": "off",
+            "@typescript-eslint/no-use-before-define": "off",
+
+            "@typescript-eslint/consistent-type-imports": "off",
+            "@typescript-eslint/naming-convention": "off",
+            "@typescript-eslint/max-params": "off",
+
+            "@/eqeqeq": "error",
```

Comment:
is this really working?

Reply:
Yep
```
/Users/petrmakhnev/tvm-dec2/src/disasm.ts
  289:26  error  Expected '===' and instead saw '=='  @/eqeqeq

✖ 1 problem (1 error, 0 warnings)

```

Reply:
didn't know it worked that way... Thanks for sharing!